cloudevents/working-drafts/xml-format.md

```diff
@@ -0,0 +1,290 @@
+the [JSON Format specification][json-format]. A `data` element MUST be used to
+encapsulate the payload.
+
+An `xsi:type` is used to discrimate the payload type and MUST be present.
+
+The following data representations are supported:
+
+### 3.1 Binary Data
+
+Binary data MUST be carried in an element with an defined type of `xs:base64Binary`.
+
+Example:
+
+``` xml
+<data xsi:type="xs:base64Binary">.........</data>
+```
+
+### 3.2 Text Data
+
+Text MUST be carried in an element with an defined type of `xs:string`.
+
+Example:
+
+``` xml
+<data xsi:type="xs:string">This is text</data>
+```
+
+### 3.3 XML Data
+
+XML data MUST be carried in an element with a defined type of `xs:any`
+allowing a single child XML element (with any required namespace definitions)
+to be represented.
+
+Example:
+
+``` xml
+<data xsi:type="xs:any">
+    <myData>
+        ....
+    </myData>  
+</data>
+```
+
+## 4. Envelope
+
+Each CloudEvent is wholly represented as an XML element `<event>` that
+MUST carry the `specversion` as an XML attribute value.
+
+Such a representation MUST use the media type `application/cloudevents+xml`.
+
+The enveloping element contains:
+
+* A set of CloudEvent context attribute XML elements.
+* An OPTIONAL `data` XML element.
+
+Example _(XML preamble and namespace definitions omitted for brevity)_:
+
+``` xml
+<event specversion="1.0">
+    <time>2020-03-19T12:54:00-07:00</time>
+    <datacontenttype>text/plain</datacontenttype>
+    <id>000-1111-2222</id>
+    <source>urn:uuid:123e4567-e89b-12d3-a456-426614174000</source>
+    <type>SOME.EVENT.TYPE</type>
+    <myboolean xsi:type="xs:boolean">false</myboolean>
+    <data xsi:type="xs:string">Now is the winter of our discount tents...</data>
+</event>
+```
+
+## 5. XML Batch Format
+
+In the _XML Batch Format_ several CloudEvents are batched into a single XML
+element. The element comprises a list of elements in the XML Format.
+
+Although the _XML Batch Format_ builds on top of the _XML Format_, it is
+considered as a separate format: a valid implementation of the _XML Format_
+doesn't need to support it. The _XML Batch Format_ MUST NOT be used when only
+support for the _XML Format_ is indicated.
+
+An XML Batch of CloudEvents MUST use the media type
+`application/cloudevents-batch+xml`.
+
+Example _(XML preamble and namespace definitions omitted for brevity)_:
+
+``` xml
+<batch>
+    <event specversion="1.0">
+       ....
+    </event>
+    <event specversion="1.0">
+       ....
+    </event>
+    ....
+</batch>
+```
+
+An example of an empty batch of CloudEvents (typically used in a response, but also valid in a request):
+
+```xml
+<batch />
+```
+
+## 6. Examples
+
+### 6.1 CloudEvent with PNG image data
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<event xmlns="http://cloudevents.io/xmlformat/V1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
+       xmlns:xs="http://www.w3.org/2001/XMLSchema" specversion="1.0" >
```

Comment:
I don't *think* the xs alias is needed here, given that we only refer to it within the value.

Reply:
It's probably this extra isn't needed at all ... it's there more for completeness that anything else.